pkgs/sdk/server-ai/src/Config/LdAiConfig.cs

```diff
@@ -0,0 +1,159 @@
+        private readonly List<Message> _prompt;
+        private readonly Dictionary<string, object> _modelParams;
+
+
+        /// <summary>
+        /// Constructs a new builder. By default, the config will be disabled, with no prompt
+        /// messages or model parameters.
+        /// </summary>
+        public Builder()
+        {
+            _enabled = false;
+            _prompt = new List<Message>();
+            _modelParams = new Dictionary<string, object>();
+        }
+
+        /// <summary>
+        /// Adds a prompt message with the given content and role. The default role is <see cref="Role.User"/>.
+        /// </summary>
+        /// <param name="content">the content, which may contain Mustache templates</param>
+        /// <param name="role">the role</param>
+        /// <returns>a new builder</returns>
+        public Builder AddPromptMessage(string content, Role role = Role.User)
+        {
+            _prompt.Add(new Message(content, role));
+            return this;
+        }
+
+        /// <summary>
+        /// Disables the config.
+        /// </summary>
+        /// <returns>the builder</returns>
+        public Builder Disable() => SetEnabled(false);
+
+        /// <summary>
+        /// Enables the config.
+        /// </summary>
+        /// <returns>the builder</returns>
+        public Builder Enable() => SetEnabled(true);
+
+        /// <summary>
+        /// Sets the enabled state of the config based on a boolean.
+        /// </summary>
+        /// <param name="enabled">whether the config is enabled</param>
+        /// <returns>the builder</returns>
+        public Builder SetEnabled(bool enabled)
+        {
+            _enabled = enabled;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets a parameter for the model. The value may be any object.
+        /// </summary>
+        /// <param name="name">the parameter name</param>
+        /// <param name="value">the parameter value</param>
+        /// <returns>the builder</returns>
+        public Builder SetModelParam(string name, object value)
+        {
+            _modelParams[name] = value;
+            return this;
+        }
+
+        /// <summary>
+        /// Builds the LdAiConfig instance.
+        /// </summary>
+        /// <returns>a new LdAiConfig</returns>
+        public LdAiConfig Build()
+        {
+            return new LdAiConfig(_enabled, _prompt, new Meta(), _modelParams);
+        }
+    }
+
+    /// <summary>
+    /// The prompts associated with the config.
+    /// </summary>
+    public readonly IReadOnlyList<Message> Prompt;
+
+    /// <summary>
+    /// The model parameters associated with the config.
+    /// </summary>
+    public readonly IReadOnlyDictionary<string, object> Model;
+
+
+
+    internal LdAiConfig(bool enabled, IEnumerable<Message> prompt, Meta meta, IReadOnlyDictionary<string, object> model)
+    {
+        Model = model ?? new Dictionary<string, object>();
```

Comment:
Should `object` here be LDValue? For JSON production purposes, no cycles, etc? (Maybe I will learn more as I proceed.)